website/docs/r/synapse_role_assignment.html.markdown

```diff
@@ -0,0 +1,94 @@
+---
+subcategory: "Synapse"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_synapse_role_assignment"
+description: |-
+  Manages a Synapse Role Assignment.
+---
+
+# azurerm_synapse_role_assignment
+
+Allows you to Manages a Synapse Role Assignment.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_storage_account" "example" {
+  name                     = "examplestorageacc"
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  account_kind             = "StorageV2"
+  is_hns_enabled           = "true"
+}
+
+resource "azurerm_storage_data_lake_gen2_filesystem" "example" {
+  name               = "example"
+  storage_account_id = azurerm_storage_account.example.id
+}
+
+resource "azurerm_synapse_workspace" "example" {
+  name                                 = "example"
+  resource_group_name                  = azurerm_resource_group.example.name
+  location                             = azurerm_resource_group.example.location
+  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
+  sql_administrator_login              = "sqladminuser"
+  sql_administrator_login_password     = "H@Sh1CoR3!"
+}
+
+resource "azurerm_synapse_firewall_rule" "example" {
+  name                 = "AllowAll"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  start_ip_address     = "0.0.0.0"
+  end_ip_address       = "255.255.255.255"
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_synapse_role_assignment" "test" {
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  role_name            = "Sql Admin"
+  principal_id         = data.azurerm_client_config.current.object_id
+}
+```
+## Argument Reference
+
+The following arguments are supported:
+
+* `synapse_workspace_id` - (Required) The ID of the Synapse Workspace on which to create the Role Assignment. Changing this forces a new resource to be created.
+
+* `role_name` - (Required) The Role Name of the Synapse Built-In Role. Changing this forces a new resource to be created.
+
+-> **NOTE:** Currently, the Synapse built-in roles are `Workspace Admin`, `Apache Spark Admin` and `Sql Admin`.
```

Comment:
could we validate against these values?

Reply:
I have added dynamic validation in the codes. If the backend service add more built-in rules, we don't need to change the code and could give user a good hint. Do you think it's better to add these three value in the field's validation rules? If so, I will do so